File: Week03/3.3P/Clock/ClockProgram/Clock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClockProgram
{
	public class Clock
	{
        public List<Counter> _counters;
        private const int SEC_LIMIT = 60;
        private const int MIN_LIMIT = 60;
        private const int HOUR_LIMIT = 24;
        public Clock()
		{
			 _counters = new List<Counter>();

			for (int i = 0; i < 3; i++)
            {
				string name_counter = "Counter " + i.ToString();
				Counter counter = new Counter(name_counter);
				_counters.Add(counter);
            }
		}

        //method

        public void Tick()
        {
            _counters[2].Increment();

            if (_counters[2].Tick == SEC_LIMIT)
            {
                _counters[2].Reset();
                _counters[1].Increment();
            }
            if (_counters[1].Tick == MIN_LIMIT)
            {
                _counters[1].Reset();
                _counters[0].Increment();
            }
            if (_counters[0].Tick == HOUR_LIMIT)
            {
                _counters[0].Reset();
            }
        }

        public void TimePrinting()
        {
            string hours = _counters[0].Tick.ToString("00");
            string minutes = _counters[1].Tick.ToString("00");
            string seconds = _counters[2].Tick.ToString("00");

            string time = hours + ":" + minutes + ":" + seconds;
            Console.WriteLine("Current Clock: {0}", time);
        }

        public string currentTime()
        {
            string hours = _counters[0].Tick.ToString("00");
            string minutes = _counters[1].Tick.ToString("00");
            string seconds = _counters[2].Tick.ToString("00");

            string time = hours + ":" + minutes + ":" + seconds;
            return time;
        }
    }
}

File: Week11/11.1P/clock.py
from counter import Counter

class Clock:
    SEC_LIMIT = 60
    MIN_LIMIT = 60
    HOUR_LIMIT = 24
    list_Of_Counter = []
    def __init__(self):
        for i in range(3):
            counter_name = "Counter " + str(i)
            counter = Counter(counter_name)
            self.list_Of_Counter.append(counter)

    def Tick(self):
        self.list_Of_Counter[2].Increment()

        if self.list_Of_Counter[2].Tick() == self.SEC_LIMIT:
            self.list_Of_Counter[2].Reset()
            self.list_Of_Counter[1].Increment()

        if self.list_Of_Counter[1].Tick() == self.MIN_LIMIT:
            self.list_Of_Counter[1].Reset()
            self.list_Of_Counter[0].Increment()

        if self.list_Of_Counter[0].Tick() == self.HOUR_LIMIT:
            self.list_Of_Counter[0].Reset()

    def TimePrinting(self):
        hour = str(self.list_Of_Counter[0].Tick()).rjust(2, '0')
        minute = str(self.list_Of_Counter[1].Tick()).rjust(2, '0')
        second = str(self.list_Of_Counter[2].Tick()).rjust(2, '0')
        time = hour + ":" + minute + ":" + second
        print("Current time: " + time)
        return
